### PR TITLE
Imported image name

### PR DESCRIPTION
**Pull request by oscaralmstrom**
New layers created by importing images will now have that image's file name as LayerName (extensions removed (.jpg, .png etc)).

### CODE REVIEW

File: src/main/java/chalmers/pimp/controller/components/ImageChooser.java [hunk 1]
```diff
@@ -43,8 +48,34 @@ public PixelData openDialog(Window window) throws IOException {
...
+   * @param resentFile the {@code File} from which the name comes from.
+   */
+  private void setResentFileName(File resentFile) {
+    List<String> temp = new ArrayList<>(Arrays.asList(resentFile.getName().split("\\.")));
+    temp.remove((temp.size() - 1));
+    StringBuilder returnString = new StringBuilder();
+    for (String s : temp) {
+      returnString.append(s);
+    }
+    resentFileName = returnString.toString();
```
Comment (molleer) [inline src/main/java/chalmers/pimp/controller/components/ImageChooser.java]: This means "image.png" => "image" which is correct but it also means "image.firstimage.1.png" => "imagefirstimage1" which is false.

Comment (molleer) [reply src/main/java/chalmers/pimp/controller/components/ImageChooser.java]: I recommend to insert this code from line 65
```
return resentfile.getName().replace("." + temp.getLast(), ""));
```

**Comment by albin-johansson** [comment]
The word you’re thinking of is recent, not resent.

File: src/main/java/chalmers/pimp/controller/components/ImageChooser.java [hunk 2]
```diff
@@ -43,8 +48,34 @@ public PixelData openDialog(Window window) throws IOException {
...
+  }
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/controller/components/ImageChooser.java]: There should probably be a service for obtaining the name of a file, excluding the file extension.

File: src/main/java/chalmers/pimp/model/canvas/layer/Raster.java [hunk 3]
```diff
@@ -34,6 +34,20 @@
...
+   *
+   * @param pixelData the pixel data that will be copied.
+   * @param pixelDataName the name of the new raster.
+   * @throws NullPointerException if the supplied pixel data is {@code null}.
+   */
+  Raster(PixelData pixelData, String pixelDataName) {
+    this.pixelData = Objects.requireNonNull(pixelData);
+    layerDelegate = new LayerDelegate(LayerType.RASTER);
+    layerDelegate.setName(pixelDataName);
   }
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/model/canvas/layer/Raster.java]: Is it okay to supply a null pixel data name?

File: src/main/java/chalmers/pimp/service/ImageImportService.java [hunk 4]
```diff
@@ -29,4 +32,15 @@ public static Image importImage(File file) throws IOException {
...
+  /**
+   * Returns the given file's filename (without extensions).
+   *
+   * @param file the file to have it's name returned.
+   * @return the name of the supplied file.
+   */
+  public static String getFileNameFromFile(File file) {
+    List<String> temp = new ArrayList<>(Arrays.asList(file.getName().split("\\.")));
+    return file.getName().replace("." + temp.get(temp.size() - 1), "");
+  }
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/service/ImageImportService.java]: Looks good but should this really be in ImageImportService? (Also what happens when the supplied file is null?)

File: src/main/java/chalmers/pimp/model/canvas/layer/LayerFactory.java [hunk 5]
```diff
@@ -35,6 +35,19 @@ public static ILayer createRasterLayer(PixelData pixelData) {
...
   }
 
+  /**
+   * Creates and returns a raster layer that holds basic raster data. The created layer is based on
+   * the supplied pixel data and a supplied name.
+   *
+   * @param pixelData the pixel data which holds the data that will be copied.
+   * @param pixelDataName the name of the pixel data that will be copied.
+   * @return a layer that holds raster data, copied from the supplied pixel data instance.
+   * @throws NullPointerException if the supplied pixel data is {@code null}.
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/model/canvas/layer/LayerFactory.java]: Tiny detail, the @throws tag should be updated.

File: src/main/java/chalmers/pimp/service/ImageImportService.java (hunk 4 above)
Comment (oscaralmstrom) [reply src/main/java/chalmers/pimp/service/ImageImportService.java]: Could make new class for this, but the whole method is pretty dependant on there being a "imported file". Just a matter of preference, will do as Merge-Master commands :neckbeard:

Comment (albin-johansson) [reply src/main/java/chalmers/pimp/service/ImageImportService.java]: It’s just that a File instance doesn’t have to point to an image

**Pull request merged**